### PR TITLE
Fix typing issues for typed Scrapy.

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

File: scrapy_zyte_api/responses.py [hunk 1]
```diff
@@ -188,8 +192,8 @@ def _process_response(
 
     if api_response.get("httpResponseHeaders") and api_response.get("httpResponseBody"):
         response_cls = responsetypes.from_args(
-            headers=api_response["httpResponseHeaders"],
-            url=api_response["url"],
+            headers=cast(List[Dict[str, str]], api_response["httpResponseHeaders"]),
```
Comment (wRAR) [inline scrapy_zyte_api/responses.py:195]: This produces an error which looks valid: `api_response["httpResponseHeaders"]` is a list of dicts with "name" and "value" keys while `ResponseTypes.from_headers` expects a `Mapping[bytes, bytes]` (not even strings).

**Review by Gallaecio — APPROVED** [review]

File: scrapy_zyte_api/responses.py (hunk 1 above)
Comment (Gallaecio) [reply scrapy_zyte_api/responses.py:195]: We’ll need to address that.

**Comment by wRAR** [comment]
Scrapy < 2.1 has a hidden bug, accidentally fixed in https://github.com/scrapy/scrapy/pull/4481: `scrapy.responsetypes.ResponseTypes.from_headers()` expects `headers` keys to be case-insensitive so it can't work with simple dicts. Looks like we need to skip a couple of tests on Scrapy 2.0.1.

**Comment by wRAR** [comment]
OTOH I think this breakage will also happen in user code. Unless we want to drop Scrapy 2.0.1, we need to use `scrapy.http.Headers` here.

File: scrapy_zyte_api/addon.py [hunk 2]
```diff
@@ -74,7 +76,7 @@ def update_settings(self, settings: BaseSettings) -> None:
             settings.set(
                 "REQUEST_FINGERPRINTER_CLASS",
                 "scrapy_zyte_api.ScrapyZyteAPIRequestFingerprinter",
-                settings.getpriority("REQUEST_FINGERPRINTER_CLASS"),
+                cast(int, settings.getpriority("REQUEST_FINGERPRINTER_CLASS")),
```
Comment (kmike) [inline scrapy_zyte_api/addon.py:79]: is it a Scrapy typing issue that getpriority doesn't annotated as returning int?

Comment (wRAR) [reply scrapy_zyte_api/addon.py:79]: It can be None. IIRC I decided that a cast is more simple than extracting the var and checking it.

Comment (kmike) [reply scrapy_zyte_api/addon.py:79]: Is it an error if the priority is None? Or is it guaranteed we'll get an int? Sorry, I'm a bit confused about the priority :) If it's None, wouldn't settings.set allow it? If no, how would you get None in a first place?

Comment (wRAR) [reply scrapy_zyte_api/addon.py:79]: `getpriority()` returns None for unset settings.

File: tests/test_sessions.py [hunk 3]
```diff
@@ -393,7 +393,7 @@ class UseChecker(ConstantChecker):
     def check(self, response: Response, request: Request) -> bool:
         if response.meta.get(SESSION_INIT_META_KEY, False) is True:
             return True
-        return super().check(request, response)
+        return super().check(response, request)
```
Comment (Gallaecio) [inline tests/test_sessions.py:396]: :open_mouth:

Comment (wRAR) [reply tests/test_sessions.py:396]: Yeah.

It's a test though.

**Pull request merged**